jdaviz/app.py

```diff
@@ -168,6 +168,11 @@ def __init__(self, configuration=None, *args, **kwargs):
         self._application_handler = JupyterApplication(
             settings={'new_subset_on_selection_tool_change': True})
 
+        # Add a reference to this application to the Glue session object. This
+        # allows the jdaviz Application object to then be accessed via e.g.
+        # viewer.session.jdaviz_app
+        self._application_handler.session.jdaviz_app = self
```

Comment:
But why can't I simply assign that to `viewer.session` at viewer creation step -- why won't that work? I don't get it.

Reply:
It looked like you might not have set it exactly at viewer creation time, it might have been slightly after so maybe it wasn't happening quite soon enough - but in any case we should set it once globally rather than in each viewer, the session is a global variable not per viewer.